nixos/modules/virtualisation/qemu-vm.nix

```diff
@@ -99,10 +106,14 @@ let
         # A writable boot disk can be booted from automatically.
         ${qemu}/bin/qemu-img create -f qcow2 -b ${bootDisk}/disk.img $TMPDIR/disk.img || exit 1
 
+        NIX_EFI_VARS=$(readlink -f ''${NIX_EFI_VARS:-${cfg.efiVars}})
+
         ${if cfg.useEFIBoot then ''
-          # VM needs a writable flash BIOS.
-          cp ${bootDisk}/bios.bin $TMPDIR || exit 1
-          chmod 0644 $TMPDIR/bios.bin || exit 1
+          # VM needs writable EFI vars
+          if ! test -e "${cfg.efiVars}"; then
```

Comment:
Why does this test `cfg.efiVars` but then write to `NIX_EFI_VARS`?

Reply:
Just an oversight on my part. Fixed.